src/BinSkim.Driver/RoslynAnalyzer/ILOperations.cs

```diff
@@ -0,0 +1,485 @@
+    {
+        public PointerIndirectionReferenceExpression(IExpression pointer)
+        {
+            Pointer = pointer;
+        }
+
+        public IExpression Pointer { get; }
+        public override ITypeSymbol ResultType => ((IPointerTypeSymbol)Pointer.ResultType).PointedAtType;
+        public override OperationKind Kind => OperationKind.PointerIndirectionReferenceExpression;
+    }
+
+    internal class LocalReferenceExpression : ReferenceExpression, ILocalReferenceExpression
+    {
+        public LocalReferenceExpression(ILocalSymbol local)
+        {
+            Local = local;
+        }
+
+        public ILocalSymbol Local { get; }
+        public override ITypeSymbol ResultType => Local.Type;
+        public override OperationKind Kind => OperationKind.LocalReferenceExpression;
+    }
+
+    internal class ParameterReferenceExpression : ReferenceExpression, IParameterReferenceExpression
+    {
+        public ParameterReferenceExpression(IParameterSymbol parameter)
+        {
+            Parameter = parameter;
+        }
+
+        public IParameterSymbol Parameter { get; }
+        public override ITypeSymbol ResultType => Parameter.Type;
+        public override OperationKind Kind => OperationKind.ParameterReferenceExpression;
+    }
+
+    internal sealed class InstanceReferenceExpression : ParameterReferenceExpression, IInstanceReferenceExpression
+    {
+        public InstanceReferenceExpression(IMethodSymbol method)
+            : base(null) // FEEDBACK: There no public API to get the this parameter. 
+        {                // Do we need IInstanceReferenceExpression to be IParameterReferenceExpresison?
+            ResultType = method.ContainingType;
+        }
+
+        public override ITypeSymbol ResultType { get; }
+        public override OperationKind Kind => OperationKind.InstanceReferenceExpression;
+
+        bool IInstanceReferenceExpression.IsExplicit => true;
+    }
+
+    internal abstract class MemberReferenceExpression : ReferenceExpression, IMemberReferenceExpression
+    {
+        protected MemberReferenceExpression(IExpression instance)
+        {
+            Instance = instance;
+        }
+
+        public IExpression Instance { get; }
+    }
+
+    internal sealed class FieldReferenceExpression : MemberReferenceExpression, IFieldReferenceExpression
+    {
+        public FieldReferenceExpression(IExpression instance, IFieldSymbol field)
+            : base(instance)
+        {
+            Field = field;
+        }
+
+        public IFieldSymbol Field { get; }
+
+        public override OperationKind Kind => OperationKind.FieldReferenceExpression;
+        public override ITypeSymbol ResultType => Field.Type;
+    }
+
+    // FEEDBACK: This does not appear to be used implemented by VB or C#, but seems
+    // like what I might  need for ldftn and ldvirtfn. Do I have the interpretation right?
```

Comment:
Yes. If VB/C# don't presently implement IMemberReferenceExpression, that's an oversight.